src/theory/evaluator.cpp

```diff
@@ -99,6 +100,177 @@ EvalResult Evaluator::evalInternal(TNode n,
 
       switch (currNodeVal.getKind())
       {
+        case kind::CONST_RATIONAL:
+        {
+          const Rational& r = currNodeVal.getConst<Rational>();
+          Assert(r.getDenominator() == 1);
```

Comment:
This is still assuming that we have integers?  I'd remove this assertion and change `r.getNumerator()` -> 'r' below. It looks like the below code is implicitly casting Integer to Rational.

Reply:
Uh.... yup.

Reply:
This should be fixed.